plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/concurrent/XRequestManager.java

```diff
@@ -74,6 +74,25 @@ synchronized public void shutdown() {
 		return submit(new XWriteRequest<>(this, description, nonCancellable, cancellable, cancel()));
 	}
 
+	/**
+	 * Allows to join all requests that have been submitted so far.
+	 *
+	 * @return a future that can be awaited
+	 */
+	synchronized public CompletableFuture<Void> allRequests() {
+		CompletableFuture<?>[] cfs;
+		synchronized (this) {
```

Comment:
Considering the method is declared synchronized, does this additional synchronized block have any effect?